package/MDAnalysis/analysis/base.py

```diff
@@ -334,63 +404,303 @@ def _setup_frames(self, trajectory, start=None, stop=None, step=None,
+
+        Parameters
+        ----------
+        backend : Union[str, BackendBase]
+            backend to be used:
+               - ``str`` is matched to a builtin backend (one of "serial",
+                 "multiprocessing" and "dask")
+               - ``BackendBase`` subclass is checked for the presence of
+                 an :meth:`apply` method
+        n_workers : int
+            positive integer with number of workers (processes, in case of
+            built-in backends) to split the work between
+        unsupported_backend : bool, optional
+            if you want to run your custom backend on a parallelizable class
+            that has not been tested by developers, by default ``False``
+
+        Returns
+        -------
+        BackendBase
+            instance of a ``BackendBase`` class that will be used for computations
+
+        Raises
+        ------
+        ValueError
+            if :attr:`parallelizable` is set to ``False`` but backend is
+            not ``serial``
+        ValueError
+            if :attr:`parallelizable` is ``True`` and custom backend instance is used
+            without specifying ``unsupported_backend=True``
+        ValueError
+            if your trajectory has associated parallelizable transformations
+            but backend is not serial
+        ValueError
+            if ``n_workers`` was specified twice -- in the run() method and durin
+            ``__init__`` of a custom backend
+        ValueError
+            if your backend object instance doesn't have an ``apply`` method
+
+
+        .. versionadded:: 2.8.0
+        """
+        builtin_backends = {
+            "serial": BackendSerial,
+            "multiprocessing": BackendMultiprocessing,
+            "dask": BackendDask
+        }
+
+        backend_class = builtin_backends.get(backend, backend)
+        supported_backend_classes = [
+            builtin_backends.get(b)
+            for b in self.get_supported_backends()
+        ]
+
+        # check for serial-only classes
+        if not self.parallelizable and backend_class is not BackendSerial:
+            raise ValueError(f"Can not parallelize class {self.__class__}")
+
+        # make sure user enabled 'unsupported_backend=True' for custom classes
+        if (not unsupported_backend and self.parallelizable
+                and backend_class not in supported_backend_classes):
+            raise ValueError((
+                f"Must specify 'unsupported_backend=True'"
+                f"if you want to use a custom {backend_class=} for {self.__class__}"
+            ))
+
+        # check for the presence of parallelizable transformations
+        if backend_class is not BackendSerial and any(
+                not t.parallelizable
+                for t in self._trajectory.transformations):
+            raise ValueError((
+                "Trajectory should not have "
+                "associated parallelizable transformations"))
```

Comment:
```suggestion
                "associated unparallelizable transformations"))
```

@marinegor please double check the correction here.